src/Logging/LoggingTrait.php

```diff
@@ -0,0 +1,166 @@
+<?php
+/**
+ * Copyright 2024 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Auth\Logging;
+
+use Psr\Log\LogLevel;
+
+trait LoggingTrait
+{
+    /**
+     * @param LogEvent $event
+     */
+    private function logRequest(LogEvent $event): void
+    {
+        $debugEvent = [
+            'timestamp' => $event->timestamp,
+            'severity' => strtoupper(LogLevel::DEBUG),
+            'clientId' => $event->clientId,
+            'requestId' => $event->requestId ?? null,
+        ];
+
+        $debugEvent = array_filter($debugEvent, fn ($value) => !is_null($value));
```

Comment:
i was going to suggest using the default behavior of `array_filter`, so you don't need to provide the second argument, but I see that [it behaves as a call to `empty`](https://www.php.net/manual/en/function.array-filter.php), which is not quite the same as you have here. Is that intentional, that you'd like to capture values of `0`? if so then I think what you have here is great. If not, we can just use `array_filter` without any arguments.

(same question applies to the calls below to `array_filter`)

Reply:
Yeah, this is intentional. The first version was using the default behaviour of `filter` but whenever there was a value of `0` or `false` it was getting removed, so I changed it.